contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,259 @@ doc: |
+      enumeration: [undefined, cartesian]
+    handedness(NX_CHAR):
+      doc: |
+        Handedness of coordinate system.
+      enumeration: [right_handed, left_handed]
+    origin(NX_CHAR):
+      doc: |
+        Location of the origin of the processing_reference_frame.
+        This specifies the location Xp = 0, Yp = 0, Zp = 0.
+        Assume regions-of-interest in this reference frame form a
+        rectangle or cuboid.
+        Edges are interpreted by inspecting the direction of their
+        outer unit normals (which point either parallel or antiparallel)
+        along respective base vector direction of the reference frame.
+      enumeration: [undefined, front_top_left, front_top_right, front_bottom_right, front_bottom_left, back_top_left, back_top_right, back_bottom_right, back_bottom_left]
+    x_alias(NX_CHAR):
+      doc: |
+        Name or alias assigned to the x-axis base vector,
+        e.g. rolling direction.
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of
+        the processing_reference_frame. We assume the configuration
+        is inspected by looking towards the sample surface from a position
+        that is located behind the detector.
+      enumeration: [undefined, north, east, south, west, in, out]
+    y_alias(NX_CHAR):
+      doc: |
+        Name or alias assigned to the y-axis base vector,
+        e.g. transverse direction.
+    y_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing y-axis base vector of
+        the processing_reference_frame. We assume the configuration
+        is inspected by looking towards the sample surface from a position
+        that is located behind the detector. For further information consult
+        also the help info for the xaxis_direction field.
+      enumeration: [undefined, north, east, south, west, in, out]
+    z_alias(NX_CHAR):
+      doc: |
+        Name or alias assigned to the z-axis base vector,
+        e.g. normal direction.
+    z_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing z-axis base vector of
+        the processing_reference frame. We assume the configuration
+        is inspected by looking towards the sample surface from a position
+        that is located behind the detector. For further information consult
+        also the help info for the xaxis_direction field.
+      enumeration: [undefined, north, east, south, west, in, out]
+  sample_reference_frame(NXcoordinate_system):
+    doc: |
+      Details about the sample/specimen reference frame.
+    type(NX_CHAR):
+      doc: |
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+        The reference frame for the sample surface reference is used for
+        identifying positions on a (virtual) image which is formed by
+        information collected from an electron beam scanning the
```

Comment:
This is much too specific (to EM) for this base class, this should either be made more general or go to the appdef.

Reply:
Same argument as already detailed, in this case I think I should revert it to NXcoordinate_system_SOMETHING but neither em nor em_ebsd would be correct because also in sample preparation you could already by virtue of the design of the experiment prepare your specimen physically using these conventions in mind. Unfortunately with area A we hadnt had the discussion neither to such level of detail nor to something fruitful which is why right now there is not even a custom schema in NOMAD for how you have polished your specimen. I agree with you that the docstring is indeed mentioning EM currently explicitly and this is a great comment insofar as really I should have cleaned that while recategorizing that concept (and I guess a few others) but that is the point, the concept here is specifically explaining using references for instances of concepts available in the field where the definition is used and indeed I agree with you that makes a good point to place it into an appdef but then this gets longer and I am sure that NXem will also at some point need to specialize to NXem_SOMETHING_MORE_SPECIFIC and then one would have to copy the concepts over again with exactly that docstring or have an even deeper inheritance chain where NXem_SOMETHING extends NXem ...

I feel it is really tricky to stick always to this idea that we should have an as minimal number of concepts as possible while in several de facto existent OWL-expressed artifacts and instantiated example people do not even follow the idea of a tree.

Reply:
Why can't you place it into `NXem `and then make `NXem_SOMETHING_MORE_SPECIFIC` extend NXem? This is the typical support application definition extension, we are using this also for `NXxps `(as extension of `NXmpes`). 

Then you put these coordinate systems in `NXem` once and `NXem_SOMETHING_MORE_SPECIFIC` just uses them without needing to copy them.